pwndbg/commands/context.py

```diff
@@ -352,9 +352,76 @@ def context(subcontext=None):
             out.flush()
 
 
+pwndbg.config.Parameter('show-compact-regs', False, 'whether to show a compact register view')
+pwndbg.config.Parameter('show-compact-regs-align', 20, 'the number of characters reserved for each register and value')
+pwndbg.config.Parameter('show-compact-regs-space', 4, 'the number of characters separating each register')
+
+
+def compact_regs(regs, width):
+    align = int(pwndbg.config.show_compact_regs_align)
+    space = int(pwndbg.config.show_compact_regs_space)
+    result = []
+
+    def calculate_padding(length):
+        return 0 if length % align == 0 else (align - (length % align))
+
+    def get_text_length(text):
+        '''Returns the text length while ignoring color-code escape sequences.'''
+        length = 0
+
+        escape_sequence = False
+        for char in text:
+            if escape_sequence:
+                escape_sequence = (char != 'm')
+            else:
+                escape_sequence = (char == '\033')
+                length += 0 if escape_sequence else 1
+
+        return length
+
+    line = ''
+    line_length = 0
+    for reg in regs:
+        reg_length = get_text_length(reg)
+
+        # Length of line with space and padding is required for fitting the
+        # register string onto the screen / display
+        line_length_with_padding = line_length
+        line_length_with_padding += space if line_length != 0 else 0
+        line_length_with_padding += calculate_padding(line_length_with_padding)
+
+        # When element does not fully fit, then start a new line
+        if line_length_with_padding + max(reg_length, align) > width:
+            result.append(line)
+
+            line = ''
+            line_length = 0
+            line_length_with_padding = 0
+
+        # Add padding
+        if line != '':
+            line += ' ' * (line_length_with_padding - line_length)
+
+        line += reg
+        line_length = line_length_with_padding + reg_length
+
+    if line != '':
+        result.append(line)
+
+    return result
+
+
 def context_regs(target=sys.stdout, with_banner=True, width=None):
     banner = [pwndbg.ui.banner("registers", target=target, width=width)]
-    return banner + get_regs() if with_banner else get_regs()
+    regs = get_regs()
+
+    if pwndbg.config.show_compact_regs:
+        if width is None:
+            _height, width = pwndbg.ui.get_window_size(target=target)
```

Comment:
shall we move this width condition to the top, so in case of `width=None` we do not fetch this twice?

(if `width=None` the `pwndbg.ui.banner(...)` called in first line of this function also gets the window size: https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/ui.py#L38-L39)

Reply:
Ok, this is a good idea.